packages/core/ios/Sources/DdSdk.swift

```diff
@@ -22,7 +22,16 @@ class RNDdSdk: NSObject {
 
     @objc(initialize:withResolver:withRejecter:)
     func initialize(configuration: NSDictionary, resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
-        nativeInstance.initialize(configuration: configuration.asDdSdkConfiguration())
+        // Datadog SDK init needs to happen on the main thread: https://github.com/DataDog/dd-sdk-reactnative/issues/198
+        // This is safe because the main queue will never be blocked waiting for the sharedQueue
+        // Adding an extra check in case sharedQueue is ever changed to be DispatchQueue.main.
+        if Thread.isMainThread {
+            nativeInstance.initialize(configuration: configuration.asDdSdkConfiguration())
+        } else {
+            DispatchQueue.main.sync {
+                nativeInstance.initialize(configuration: configuration.asDdSdkConfiguration())
+            }
+        }
         resolve(nil)
     }
```

Comment:
The extra check is required to avoid deadlock, I'm not familiar with RN threading but since the method signature has callbacks it seems safer to do an `async` dispatch here:
```suggestion
@objc(initialize:withResolver:withRejecter:)
func initialize(configuration: NSDictionary, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) -> Void {
    DispatchQueue.main.async {
        self.nativeInstance.initialize(configuration: configuration.asDdSdkConfiguration())
        resolve(nil)
    }
}
```

Would that work?

Reply:
Yeah, should be totally fine. I didn’t sync to keep the sharedQueue busy, but it’s unlikely any other work would happen in parallel to this.

The extra check is unneeded today, but if a developer ever changed the sharedQueue to DispatchQueue.main then it prevents the deadlock in that case.